packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/RealmAnyTests.kt

```diff
@@ -421,6 +422,70 @@ class RealmAnyTests {
+                    assertEquals(RealmAny.create(1), RealmAny.create(1.toInt()))
+                    assertEquals(RealmAny.create(1), RealmAny.create(1.toLong()))
+                    assertNotEquals(RealmAny.create(1), RealmAny.create(2))
+                }
+                RealmAny.Type.BOOL -> {
+                    assertEquals(RealmAny.create(true), RealmAny.create(true))
+                    assertNotEquals(RealmAny.create(true), RealmAny.create(false))
+                }
+                RealmAny.Type.STRING -> {
+                    assertEquals(RealmAny.create("Realm"), RealmAny.create("Realm"))
+                    assertNotEquals(RealmAny.create("Realm"), RealmAny.create("Not Realm"))
+                }
+                RealmAny.Type.BINARY -> {
+                    assertEquals(
+                        RealmAny.create(byteArrayOf(1, 2)), RealmAny.create(byteArrayOf(1, 2))
+                    )
+                    assertNotEquals(
+                        RealmAny.create(byteArrayOf(1, 2)), RealmAny.create(byteArrayOf(2, 1))
+                    )
+                }
+                RealmAny.Type.TIMESTAMP -> {
+                    val now = RealmInstant.now()
+                    assertEquals(RealmAny.create(now), RealmAny.create(now))
+                    assertNotEquals(RealmAny.create(RealmInstant.from(1, 1)), RealmAny.create(now))
+                }
+                RealmAny.Type.FLOAT -> {
+                    assertEquals(RealmAny.create(1.5f), RealmAny.create(1.5f))
+                    assertNotEquals(RealmAny.create(1.2f), RealmAny.create(1.3f))
+                }
+                RealmAny.Type.DOUBLE -> {
+                    assertEquals(RealmAny.create(1.5), RealmAny.create(1.5))
+                    assertNotEquals(RealmAny.create(1.2), RealmAny.create(1.3))
+                }
+                RealmAny.Type.DECIMAL128 -> {
+                    assertEquals(RealmAny.create(Decimal128("1E64")), RealmAny.create(Decimal128("1E64")))
+                    assertNotEquals(RealmAny.create(Decimal128("1E64")), RealmAny.create(Decimal128("-1E64")))
+                }
+                RealmAny.Type.OBJECT_ID -> {
+                    val value = ObjectId()
+                    assertEquals(RealmAny.create(value), RealmAny.create(value))
+                    assertNotEquals(RealmAny.create(ObjectId()), RealmAny.create(value))
+                }
+                RealmAny.Type.UUID -> {
+                    val value = RealmUUID.random()
+                    assertEquals(RealmAny.create(value), RealmAny.create(value))
+                    assertNotEquals(RealmAny.create(RealmUUID.random()), RealmAny.create(value))
+                }
+                RealmAny.Type.OBJECT -> {
+                    val realmObject = Sample()
+                    assertEquals(RealmAny.create(realmObject), RealmAny.create(realmObject))
+                    assertNotEquals(RealmAny.create(Sample()), RealmAny.create(realmObject))
```

Comment:
I would also add a test that exercises `if (other.clazz != this.clazz) return false`